monkey/infection_monkey/exploit/zerologon.py

```diff
@@ -0,0 +1,743 @@
+"""
+Zerologon, CVE-2020-1472
+Implementation based on https://github.com/dirkjanm/CVE-2020-1472/ and https://github.com/risksense/zerologon/.
+"""
+
+import cmd
+import io
+import logging
+import ntpath
+import os
+import sys
+import time
+import traceback
+from binascii import unhexlify
+
+import impacket
+from impacket.dcerpc.v5 import epm, nrpc, transport
+from impacket.dcerpc.v5.dcom import wmi
+from impacket.dcerpc.v5.dcomrt import DCOMConnection
+from impacket.dcerpc.v5.dtypes import NULL
+from impacket.examples.secretsdump import (LocalOperations, LSASecrets,
+                                           NTDSHashes, RemoteOperations,
+                                           SAMHashes)
+from impacket.smbconnection import SMBConnection
+
+from common.utils.exploit_enum import ExploitType
+from infection_monkey.exploit.HostExploiter import HostExploiter
+from infection_monkey.network.zerologon_fingerprint import ZerologonFinger
+
+LOG = logging.getLogger(__name__)
+
+
+_orig_stdout = None
+_new_stdout = None
+
+
+def _set_stdout_to_in_memory_text_stream():
+    # set stdout to in-memory text stream, to capture info that would otherwise be printed
+    global _orig_stdout, _new_stdout
```

Comment:
Have you investigated any more elegant solutions than modifying globals? Can't we redirect the outputs of LOG by modifying the logger? `logging.getLogger("imported_module")` and then set our own custom handler that pushes logs to a variable? Useful example: https://stackoverflow.com/questions/37944111/python-rolling-log-to-a-variable

Reply:
We don't need to redirect the outputs of LOG. Somewhere inside impacket, it prints everything it finds to stdout, which is what we're redirecting to a variable.